lib/ui/painting.dart

```diff
@@ -2762,9 +2757,34 @@ class ColorFilter {
     }
   }
 
+  @override
+  bool operator ==(Object other) {
+    return other is ColorFilter
```

Comment:
See style guide

https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#common-boilerplates-for-operator-and-hashcode

Reply:
But also - why do we want this or hashcode?

Reply:
See https://github.com/flutter/engine/pull/13711

If they don't support == then we always rebuild any tree they are used in (though ColorFilter const-ness may reduce some of that).

Reply:
Ahh okm. Just need the runtimeTyoe check here and below

Reply:
Do the private `ImageFilter` classes need the runtime check as well? They don't have subclasses so I guess the `is` check should suffice?

Reply:
It's best to just add it now rather than forgetting later if someone subclasses it. Also keeps the code consistent when refactoring these things.